### ISSUE TITLE
UnicodeDecodeError when trying to upload an file

### ISSUE THREAD

**Issue by sdushantha** [sdushantha/qr-filetransfer#24]
I am getting an error when I run

```
$ python3 qr-filetransfer -r ~/Downloads/ -debug
```
```
10.0.0.4 - - [01/May/2019 09:27:05] "GET / HTTP/1.1" 200 -
10.0.0.4 - - [01/May/2019 09:27:05] "GET /favicon.ico HTTP/1.1" 200 -
----------------------------------------
Exception happened during processing of request from ('10.0.0.4', 53764)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 720, in __init__
    self.handle()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/server.py", line 426, in handle
    self.handle_one_request()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/server.py", line 414, in handle_one_request
    method()
  File "qr-filetransfer", line 64, in do_POST
    r, info = self.deal_post_data()
  File "qr-filetransfer", line 108, in deal_post_data
    fn = re.findall(r'Content-Disposition.*name="file"; filename="(.*)"', line.decode())
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 71: invalid start byte
----------------------------------------
```

@npes87184 Do you know anything about this?

**Comment by npes87184** [comment]
Hi, @sdushantha 

Do you tried to upload something?
Can you please provide more detail? For example, the filename, how to reproduce.

Thank you!

**Comment by sdushantha** [comment]
@npes87184 Here is what I did:

I ran this command
```
$ python3 qr-filetransfer -r ~/Downloads/ -debug
```

Then, I opened the url on my computer browser and selected an image with this file name:
Filename: `10-super-comfy-wallpapers.-vol.-2-3-1600×900.jpg`

When I clicked upload, I got this error:
```
Exception happened during processing of request from ('10.0.0.4', 56201)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 720, in __init__
    self.handle()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/server.py", line 426, in handle
    self.handle_one_request()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/server.py", line 414, in handle_one_request
    method()
  File "qr-filetransfer", line 64, in do_POST
    r, info = self.deal_post_data()
  File "qr-filetransfer", line 108, in deal_post_data
    fn = re.findall(r'Content-Disposition.*name="file"; filename="(.*)"', line.decode())
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd7 in position 95: invalid continuation byte
```
Here is the image, if you want to try it:
https://user-images.githubusercontent.com/27065646/57017508-cff16300-6c1f-11e9-85b2-267ce7006e09.jpg

**Comment by sdushantha** [comment]
closing this because it has been fixed in #25
